engine/condition_patch.go

```diff
@@ -0,0 +1,95 @@
+package engine
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/bivas/rivi/types"
+	"github.com/bivas/rivi/util/log"
+	"github.com/waigani/diffparser"
+)
+
+type PatchCondition struct {
+	Hunk struct {
+		StartsAt int    `mapstructure:"starts-at,omitempty"`
+		Pattern  string `mapstructure:"pattern,omitempty"`
+	} `mapstructure:"hunk,omitempty"`
+}
+
+func (c *PatchCondition) IsEmpty() bool {
+	return c.Hunk.StartsAt == 0 && c.Hunk.Pattern == ""
+}
+
+func (c *PatchCondition) Match(meta types.Data) bool {
+	if c.IsEmpty() {
+		return false
+	}
+	patchConditionCounter.Inc()
+	extended, ok := meta.(types.ExtendedData)
+	if !ok {
+		lc.DebugWith(
+			log.MetaFields{log.F("condition", "PatchCondition"), log.F("issue", meta.GetShortName())},
+			"Meta data is not extended data")
+		return false
+	}
+	re, err := regexp.Compile(c.Hunk.Pattern)
```

Comment:
IMO this can  be done in a validate state of creating the rules (just once a not every time).

Reply:
You're correct. I'll open an issue to follow up on this and fix it globally.